src/lib/swapper/LifiSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,185 @@
+import type { ChainKey, LifiError, QuoteRequest, Step, TokensResponse } from '@lifi/sdk'
+import type LIFI from '@lifi/sdk'
+import { LifiErrorCode } from '@lifi/sdk'
+import { fromChainId } from '@shapeshiftoss/caip'
+import type { EvmChainId } from '@shapeshiftoss/chain-adapters'
+import type { GetEvmTradeQuoteInput, QuoteFeeData, TradeQuote } from '@shapeshiftoss/swapper'
+import { SwapError, SwapErrorType } from '@shapeshiftoss/swapper'
+import { DEFAULT_SLIPPAGE } from 'constants/constants'
+import { BigNumber, bn, bnOrZero } from 'lib/bignumber/bignumber'
+import type { LifiToolMeta } from 'lib/swapper/LifiSwapper/types'
+import {
+  DEFAULT_SOURCE,
+  MIN_AMOUNT_THRESHOLD_USD_HUMAN,
+} from 'lib/swapper/LifiSwapper/utils/constants'
+import { convertPrecision } from 'lib/swapper/LifiSwapper/utils/convertPrecision/convertPrecision'
+import { getMinimumAmountFromStep } from 'lib/swapper/LifiSwapper/utils/getMinimumAmountFromStep/getMinimumAmountFromStep'
+import { selectPortfolioCryptoBalanceByFilter } from 'state/slices/common-selectors'
+import {
+  selectAccountIdByAccountNumberAndChainId,
+  selectMarketDataById,
+} from 'state/slices/selectors'
+import { store } from 'state/store'
+
+export async function getTradeQuote(
+  input: GetEvmTradeQuoteInput,
+  lifiTokens: TokensResponse['tokens'],
+  lifiChainMap: Map<number, ChainKey>,
+  lifiToolMap: Map<string, Map<string, Map<string, LifiToolMeta>>>,
+  lifi: LIFI,
+): Promise<TradeQuote<EvmChainId>> {
+  const {
+    chainId,
+    sellAsset,
+    buyAsset,
+    sellAmountBeforeFeesCryptoBaseUnit,
+    sendMax,
+    receiveAddress,
+    accountNumber,
+  } = input
+
+  const fromLifiChainId = +fromChainId(sellAsset.chainId).chainReference
```

Comment:
Use `sellAsset` / `buyAsset` instead of `from` / `to`, which can be confused as actions (e.g `fromChainId()` method)